storage_service/locations/models/dataverse.py

```diff
@@ -112,63 +207,110 @@ def move_to_storage_service(self, src_path, dest_path, dest_space):
         """
         # TODO how to strip location path if location isn't passed in?
         # HACK strip everything that isn't a number
-        src_path = ''.join(c for c in src_path if c.isdigit())
+        src_path = "".join(c for c in src_path if c.isdigit())
         # Verify src_path has to be a number
         if not src_path.isdigit():
-            raise StorageException(_('Invalid value for src_path: %(value)s. Must be a numberic entity_id') % {'value': src_path})
+            raise StorageException(
+                _(
+                    "Invalid value for src_path: %(value)s. Must be a "
+                    "numeric entity_id"
+                )
+                % {"value": src_path}
+            )
         # Fetch dataset info
-        url = 'https://' + self.host + '/api/datasets/' + src_path
-        params = {
-            'key': self.api_key,
-        }
-        LOGGER.debug('URL: %s, params: %s', url, params)
+        url = "https://{}/api/datasets/{}".format(self.host, src_path)
```

Comment:
Do you have an opinion @sevein on what the best strategy for managing the scheme should look like? 

In the DV form, we'd specify something like this:

![image](https://user-images.githubusercontent.com/1880412/45699125-1aa9e900-bb2f-11e8-9696-22cb72f01e71.png)

So I can parse the URL and pull the scheme out of there, and if there isn't one, add `https` as a default. Does that sound okay?

Reply:
That makes sense, @ross-spencer. I recently made a similar change in the pipeline settings, there is a method that could be converted into a mixin and be used here too?

https://github.com/artefactual/archivematica-storage-service/blob/07ae87501b93f4c917f1c0a49dc76ba5c369039d/storage_service/locations/models/pipeline.py#L76-L89

In this case `scheme` defaults to `http` but I guess that you can make it a parameter.

Thanks!

Reply:
I've given it a whirl in https://github.com/artefactual/archivematica-storage-service/pull/398/commits/c9b630ba6abcc671033e0e573df65a10afaae930, does that look okay?

Reply:
Looks good to me. I hope JD is okay with this OOP-ish solution haha!

The double underscore may not be great because in Python I think they convention is to use them for magic methods, like `__repr__`, etc... I think you could just call it `_generate_dataverse_url`.

Reply:
+1 to removing one of those underscores.

Also, I don't see why the `slug` and `scheme` kwargs are needed in this method if they're never used:

    $ ack __generate_dataverse_url ./
    storage_service/locations/models/dataverse.py
    121:        url = self.__generate_dataverse_url("/api/search/")
    175:        url = self.__generate_dataverse_url(files_in_dataset_path)
    219:        url = self.__generate_dataverse_url(datasets_url)
    271:                url = self.__generate_dataverse_url(bundle_url)
    277:                url = self.__generate_dataverse_url(datafile_url)
    321:    def __generate_dataverse_url(self, slug="", scheme="https"):